File: app/src/main/java/mx/dev/franco/automusictagfixer/covermanager/ICoverRunnable.java
package mx.dev.franco.automusictagfixer.covermanager;

public interface ICoverRunnable {
    String getPath();
    /**
     * Sets the actions for each state of the PhotoTask instance.
     * @param state The state being handled.
     */
    void handleExtractionState(int state);

    /**
     * Set the cover as byte array to display.
     * @param cover
     */
    void setCover(byte[] cover);
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/covermanager/ICoverRunnable.kt
package mx.dev.franco.automusictagfixer.covermanager

interface ICoverRunnable {
    val path: String?

    /**
     * Sets the actions for each state of the PhotoTask instance.
     * @param state The state being handled.
     */
    fun handleExtractionState(state: Int)

    /**
     * Set the cover as byte array to display.
     * @param cover
     */
    fun setCover(cover: ByteArray?)
}